src/kvstore/RocksEngine.cpp

```diff
@@ -151,14 +171,46 @@ nebula::cpp2::ErrorCode RocksEngine::commitBatchWrite(std::unique_ptr<WriteBatch
   options.sync = sync;
   options.no_slowdown = !wait;
   auto* b = static_cast<RocksWriteBatch*>(batch.get());
-  rocksdb::Status status = db_->Write(options, b->data());
-  if (status.ok()) {
-    return nebula::cpp2::ErrorCode::SUCCEEDED;
-  } else if (!wait && status.IsIncomplete()) {
-    return nebula::cpp2::ErrorCode::E_WRITE_STALLED;
+  auto dataCount = b->data()->Count();
+  auto refCount = b->refdata()->Count();
+
+  rocksdb::Status status;
+  rocksdb::Status refStatus;
+  if (dataCount && refCount) {
+    status = db_->Write(options, b->data());
```

Comment:
i think you should check this status before refdb_->write()
otherwise refdb_  maybe dirty

Reply:
Good 👍